### ISSUE TITLE
Amazon EKS cluster public endpoint network access restriction via IPv4 address ranges

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#11397]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon Elastic Kubernetes Service (EKS) now allows you to restrict access to your Kubernetes cluster’s public endpoint by specifying allowed IPv4 address ranges in CIDR notation

### New or Affected Resource(s)

* [`aws_eks_cluster`](https://www.terraform.io/docs/providers/aws/r/eks_cluster.html)

### Potential Terraform Configuration

```hcl
resource "aws_eks_cluster" "example" {
  vpc_config {
    public_access_cidrs = ["203.0.113.5/32"]
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/12/amazon-eks-enables-network-access-restrictions-to-kubernetes-cluster-public-endpoints/).
[User guide](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html#w242aac13c21c17b9b3).
[AWS container services roadmap](https://github.com/aws/containers-roadmap/issues/108).

Requires [AWS SDK v1.26.7](https://github.com/aws/aws-sdk-go/releases/tag/v1.26.7):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11375

**Comment by seddarj** [comment]
Fix implemented in #11442. Feel free to comment on that if you see anything wrong there :D

**Comment by bflad** [comment]
Support for this has been merged and will release with version 2.44.0 of the Terraform AWS Provider, tomorrow. Thanks to @seddarj for the implementation. 👍
